public/app/features/variables/state/helpers.ts

```diff
@@ -161,3 +162,19 @@ export function getPreloadedState(
     },
   };
 }
+
+// function to find a query variable node in the list of all variables
+export function findVariableNodeInList(allVariables: TypedVariableModel[], nodeName: string) {
+  const variableNode = allVariables.find((v) => {
+    return v.name === nodeName;
+  });
+  return variableNode;
+}
+
+// Some variable types have a refresh options, and it might be configured to refresh on time range change
+export function isVariableOnTimeRangeConfigured(variable: TypedVariableModel) {
+  return (
+    (variable.type === 'query' || variable.type === 'datasource' || variable.type === 'interval') &&
+    variable.refresh === VariableRefresh.onTimeRangeChanged
+  );
```

Comment:
This function looks funky. For datasource and interval there is no time range refresh configuration, why handling those types here?

Reply:
Yes, I was also surprised. Currently, we don't offer a UI configuration for time range refresh for either interval or datasource variables. However, when inspecting the Dashboard JSON, I noticed that interval variables had the default configuration set to "time range," and datasource variables had the property assigned to their model. With the datasource variable, I wasn't sure if we might receive provisioned dashboards with this default setting for time range. 

So, I decided to create this function, especially because in the original function, we were checking for the `refresh` prop and were using the `as QueryVariable` cast. 🙈

Reply:
> I wasn't sure if we might receive provisioned dashboards with this default setting for time range.

Well, think this is a valid assumption, not sure if real case, but worth having this extra security. I suggest adding a comment about this to the function so that it won't get lost.